lib/dotcom_web/components/trip_planner/itinerary_group.ex

```diff
@@ -14,12 +16,31 @@ defmodule DotcomWeb.Components.TripPlanner.ItineraryGroup do
   """
   def itinerary_group(assigns) do
     ~H"""
-    <div class="mb-3 p-2 border border-2 border-slate-200">
-      <% [first | rest] = @group %>
-      <div class="text-slate-800 font-bold">Group with <%= Enum.count(@group) %> options</div>
+    <% [first | rest] = @group %>
+    <% [first_leg | rest_of_legs] = first.legs %>
+    <div class="border border-gray-lighter border-solid m-4 p-4">
+      <div class="flex flex-row mb-3">
```

Comment:
I'm on the fence about this being a whole big thing. It feels like a lot is going on in this one HEEx blob, but I'm not totally sure about the optimal way to slice and dice it.

Happy to either slice it up as part of this PR, or refactor as a follow-up.